src/test/java/lambda/part1/exercise/Exercise1.java

```diff
@@ -0,0 +1,137 @@
+                return Integer.compare(person1.getAge(), person2.getAge());
+            }
+        }
+
+        Arrays.sort(persons, new compareByAge());
+
+        assertArrayEquals(new Person[]{
+                new Person("Иван", "Мельников", 20),
+                new Person("Николай", "Зимов", 30),
+                new Person("Алексей", "Доренко", 40),
+                new Person("Артем", "Зимов", 45)
+        }, persons);
+
+        assertArrayEquals(new Person[]{
+                new Person("Иван", "Мельников", 20),
+                new Person("Николай", "Зимов", 30),
+                new Person("Алексей", "Доренко", 40),
+                new Person("Артем", "Зимов", 45)
+        }, persons);
+    }
+
+    @Test
+    public void sortPersonsByAgeUsingArraysSortAnonymousComparator() {
+        Person[] persons = getPersons();
+
+        // TODO использовать Arrays.sort
+        Arrays.sort(persons, new Comparator<Person>() {
+            @Override
+            public int compare(Person person1, Person person2) {
+                return Integer.compare(person1.getAge(), person2.getAge());
+            }
+        });
+
+        assertArrayEquals(new Person[]{
+                new Person("Иван", "Мельников", 20),
+                new Person("Николай", "Зимов", 30),
+                new Person("Алексей", "Доренко", 40),
+                new Person("Артем", "Зимов", 45)
+        }, persons);
+    }
+
+    @Test
+    public void sortPersonsByLastNameThenFirstNameUsingArraysSortAnonymousComparator() {
+        Person[] persons = getPersons();
+
+        // TODO использовать Arrays.sort
+        Arrays.sort(persons, new Comparator<Person>() {
+            @Override
+            public int compare(Person person1, Person person2) {
+                final String firstSirname = person1.getLastName();
+                final String secondSirname = person2.getLastName();
+                final String firstName = person1.getFirstName();
+                final String secondName = person2.getFirstName();
+                return firstSirname.compareTo(secondSirname)>0 ||
+                        firstSirname.compareTo(secondSirname)==0 && firstName.compareTo(secondName)>0? 1:
+                        firstSirname.compareTo(secondSirname)<0 ||
+                                firstSirname.compareTo(secondSirname)==0 && firstName.compareTo(secondName)<0? -1: 0;
+            }
+        });
+
+        assertArrayEquals(new Person[]{
+                new Person("Алексей", "Доренко", 40),
+                new Person("Артем", "Зимов", 45),
+                new Person("Николай", "Зимов", 30),
+                new Person("Иван", "Мельников", 20)
+        }, persons);
+    }
+
+    @Test
+    public void findFirstWithAge30UsingGuavaPredicate() {
+        List<Person> persons = Arrays.asList(getPersons());
+
+        // TODO использовать FluentIterable
+        Predicate<Person> isAge30 = new Predicate<Person>() {
```

Comment:
I think, you must declare a ```class``` here which will implements Guava's ```Predicate```. Then create and use it's instance.

Reply:
For what reason must I declare a class, if the class is used only once and the virtual class can be good?

Reply:
Because there are another subtask that require implementation using anonymous class. As I understand, here class declaring is meant.